client/src/components/EditProfile/PaymentTab/PaymentTab.tsx

```diff
@@ -0,0 +1,125 @@
+import useStyles from './useStyles';
+import Typography from '@material-ui/core/Typography';
+import { Box, Button, OutlinedInput, Grid } from '@material-ui/core';
+import { useState } from 'react';
+import { useFormik } from 'formik';
+import CreditCard from './CreditCard/CreditCard';
+
+interface card {
+  name: string;
+  cardNumber: string;
+  cardExpDate: string;
+  cardCVC: string;
+  cardType: string;
+}
+
+export default function PaymentTab(): JSX.Element {
+  const classes = useStyles();
+  const [savedCards, setSavedCards] = useState<card[]>([]);
+  const [showCardInput, setShowCardInput] = useState(false);
+
+  const addCard = (newCard: card) => {
+    setSavedCards([...savedCards, newCard]);
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      name: '',
+      cardNumber: '',
+      cardExpDate: '',
+      cardCVC: '',
+    },
+    onSubmit: (values) => {
+      addCard({
+        name: values.name,
+        cardNumber: values.cardNumber,
+        cardExpDate: values.cardExpDate,
+        cardCVC: values.cardCVC,
+        cardType: ['visa', 'mastercard'][Math.floor(Math.random() * 2)],
+      });
```

Comment:
If you collect the card information with stripe elements, you can also access this information from the context with the help of [`useElements` hook](https://stripe.com/docs/stripe-js/react#useelements-hook). The card brand would be automatically determined for you that way too! 🎉